ultralytics/solutions/dwell_time_analyzer.py

```diff
@@ -0,0 +1,475 @@
+                txt_color=self.analytics_text_color,
+                box_style=True,
+            )
+            offset += 30
+            if funnel_conversion is not None:
+                annotator.text(
+                    (10, offset),
+                    f"  Conversion Rate: {funnel_conversion:.2f}%",
+                    txt_color=self.analytics_text_color,
+                    box_style=True,
+                )
+            else:
+                annotator.text(
+                    (10, offset),
+                    "  Conversion Rate: N/A",
+                    txt_color=self.analytics_text_color,
+                    box_style=True,
+                )
+            offset += 30
+
+        # Display average dwell times if enabled
+        if self.enable_avg_dwell and avg_dwell_times:
+            annotator.text(
+                (10, offset),
+                "Average Dwell Times (per zone):",
+                txt_color=self.analytics_text_color,
+                box_style=True,
+            )
+            offset += 30
+            for z_name, avg_dt in avg_dwell_times.items():
+                unit = "s" if self.fps else "frames"
+                annotator.text(
+                    (10, offset),
+                    f"  {z_name}: {avg_dt:.2f}{unit}",
+                    txt_color=self.analytics_text_color,
+                    box_style=True,
+                )
+                offset += 30
+
+        im0[:] = annotator.result()
+
+    def draw_zones(self, im0: Any, current_zone_counts: Dict[str, int]) -> None:
+        """
+        Draws the defined zones on the image frame along with their labels and current counts.
+
+        Args:
+            im0 (Any): The image frame on which to draw the zones.
+            current_zone_counts (Dict[str, int]): Current counts of objects in each zone.
+        """
+        annotator = Annotator(im0, line_width=self.line_width)
+        for zone_name, polygon in self.zone_polygons.items():
+            if polygon.geom_type == "Polygon":
+                coords = list(polygon.exterior.coords)
+            else:
+                coords = list(polygon.coords)
+            coords = [(int(x), int(y)) for (x, y) in coords]
+            # Blue color for the zone
+            annotator.draw_region(
+                reg_pts=coords,
+                color=(255, 0, 0),  # BGR for Blue
+                thickness=self.line_width,
+            )
+            # Red color for the zone label
+            count_in_zone = current_zone_counts.get(zone_name, 0)
+            zone_label = f"{zone_name} | Count: {count_in_zone}"
+            if coords:
+                annotator.text(
+                    (coords[0][0], coords[0][1] - 10),
+                    zone_label,
+                    txt_color=(0, 0, 255),
+                    box_style=True,
+                )  # Red label
+        im0[:] = annotator.result()
+
+    def count(self, im0: Any) -> Any:
```

Comment:
@Muhammad-237 Hi, can we use the `count` function already defined in `object_counter.py`: https://github.com/ultralytics/ultralytics/blob/main/ultralytics/solutions/object_counter.py?

Reply:
Its goal is more about zone-based analytics (e.g., time spent in a specific area) rather than just counting objects. and the method in object_counter.py is just counting.